src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/service/ExternalTaskWorker.java

```diff
@@ -48,20 +57,31 @@ public void setupClient() {
     }
 
     public void checkIdempotency(ExternalTask externalTask, ExternalTaskService externalTaskService) {
-        Boolean isDuplicate =  externalTask.getVariable("isDuplicate");
-        if (isDuplicate == null || isDuplicate) {
+
+
+        Optional<IdempotentKeys> existingTask = idempotentKeysRepository.findById(new IdempotentId(externalTask.getVariable("idempotentKey"),"ia"));
+        if (existingTask.isPresent()) {
             Map<String, Object> processVariables = singletonMap(
                 "isDuplicate",
-                false
+                true
             );
-
-            LOGGER.info("Duplicate was hit.");
-
+            LOGGER.info("It was present!!!!!");
             externalTaskService.complete(externalTask, processVariables);
         } else {
-            externalTaskService.complete(externalTask);
-            LOGGER.info("Duplicate was true");
+            LOGGER.info("It was not present!!!!!!");
+            IdempotentKeys keys = new IdempotentKeys(
+                new IdempotentId(UUID.randomUUID().toString(), "ia"),
```

Comment:
we have to use here the new IdempotentId(externalTask.getVariable("idempotentKey"),"ia") right?